### ISSUE TITLE
3.1 and mvc are these doc still up todate?

### ISSUE THREAD

**Issue by Seabizkit** [dotnet/AspNetCore.Docs#17125]
It hard to tell what is the correct way to do things.

app.UseEndpoints(.....  vs app.UseMvc(....

if the migration process it talks about UseEndpoints but here it talks about UseMvc it is extremely hard to figure out how to implement "non default/non simple" routing if the doc are not consistent.

im trying to setup areas,
 but having a hard time trying to figure out what i need to do, 
with regards to my requirements.

more examples and more non standard samples and be consistent with how you do for x version of .net core.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 8ee6ad2f-1667-8d5c-1f36-9d0511e3720f
* Version Independent ID: 7838ccec-c8e8-9baa-2b3e-220b2af5fc58
* Content: [Routing to controller actions in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/mvc/controllers/routing?view=aspnetcore-3.1)
* Content Source: [aspnetcore/mvc/controllers/routing.md](https://github.com/dotnet/AspNetCore.Docs/blob/master/aspnetcore/mvc/controllers/routing.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-mvc**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

**Comment by guardrex** [comment]
Hello @Seabizkit ... This is covered by the cross-links on the master issue, especially https://github.com/dotnet/AspNetCore.Docs/issues/14655. I'll add your issue to that list.

**Comment by Rick-Anderson** [comment]
@Seabizkit  this is fixed in #16456
Hopefully it will be published soon.

**Comment by Seabizkit** [comment]
thanks, its much in need but i see there is a lot of changes... so not as simple as one would think LOL, but we need docs which are easier to follow. ++ 

im trying to do something which i feel the examples should cover as its most likely what someone will do but hard than its should be to figure out.

I have this which is working:
```
  endpoints.MapAreaControllerRoute(
       name: "Importer_default",
       areaName : "Importer",
      pattern: "Org/{orgName}/Importer/{controller=Dashboard}/{action=Index}/{id?}"
);
```
now im trying to "override/modify/tweek" one of the controllers (Template) in area "Importer".
the Routes above works and generates links correct: 
"Org/{orgName}/Importer/Template/Index"

but lets say for this Route i wanted to add another "Url segment" for Template only like
"Org/{orgName}/Importer/Source/Template/Index"

where i added "Source" before the Controller name, I have tried added another endpoint and have tired 
[Route("Org/{orgName}/Importer/Source/[controller]/[action]")]
on the Controller but neither make it so that 

<a asp-area="Importer" asp-controller="Template"  asp-action="Index" asp-route-id="@item.Id">test</a>

generate correctly.
